pkg/cache/v2/cache.go

```diff
@@ -51,21 +52,66 @@ type Cache interface {
 	Fetch(context.Context, Request) (*Response, error)
 }
 
+type ResponseType interface {
+	GetMarshalled() ([]*any.Any, error)
```

Comment:
`GetMarshaledResources` would probably be more accurate. This could also be made `GetDiscoveryResponse` IMO, which might even be better for the pass through.

Reply:
makes sense.